src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -42,24 +55,31 @@ public bool IsEnd
                 }
                 else
                 {
-                    var block = _block.Next;
-                    while (block != null)
-                    {
-                        if (block.Start < block.End)
-                        {
-                            return false; // subsequent block has data - IsEnd is false
-                        }
-                        block = block.Next;
-                    }
-                    return true;
+                    return IsEndMultiBlock();
```

Comment:
So realistically, this calls the non-inlined method anytime IsEnd returns true. Should we add a special case in the getter for when `_block.Next == null` since we already checked that `_index >= _block.End`?